server/src/main/java/org/elasticsearch/index/codec/tsdb/es819/ES819TSDBDocValuesProducer.java

```diff
@@ -544,6 +558,76 @@ BytesRef decode(int docNumber, int numBlocks) throws IOException {
             uncompressedBytesRef.length = end - start;
             return uncompressedBytesRef;
         }
+
+        void decodeBulk(int numBlocks, int firstDoc, int count, BlockLoader.SingletonBytesRefBuilder builder) throws IOException {
+            int remainingCount = count;
+            int nextDoc = firstDoc;
+            int blockDocOffset = 0;
+            int blockByteOffset = 0;
+            long[] offsets = new long[count + 1];
+            List<BytesRef> decompressedBlocks = new ArrayList<>();
```

Comment:
Maybe add a TODO, to account memory usage of `decompressedBlocks`? This could grow and have a significant jvm heap memory usage. 

Eventually this will be accounted, but that happens when we hand over byte[] to the builder.